### PR TITLE
[WIP] Updates chpt10 - neural networks

### PR DESCRIPTION
**Pull request by joeyklee**
Adds updates to Chapter 10

Notes:
* will only update up until 10.6 - chapter 10.7+ will be updated to use ml5
* will not include visualizing the neural network - 10.7 Neural Network Diagrams

* Do we update the quote in the beginning of the Chapter from Charlie Sheen? As we're revisiting updating quotes and refs to people w/ some rather unsavory history or dated refs etc. 

Update: latest
* @shiffman I've added up to 10.6 and is ready for you to take over. One question before I merge in to `source`: Should I remove the content from chapt.10.7 or just leave it as is until you go through your round of edits and content additions?

### CODE REVIEW

**Comment by shiffman** [comment]
> Do we update the quote in the beginning of the Chapter from Charlie Sheen? As we're revisiting updating quotes and refs to people w/ some rather unsavory history or dated refs etc.

Yes, I think this should be changed. I enjoy using humorous quotes as a friendly way for readers to enter a difficult and scary topic, but I would also like to prioritize positive messages and be thoughtful about representation. I'll think about a new one!

> Should I remove the content from chapt.10.7 or just leave it as is until you go through your round of edits and content additions?

Leave the content in and I will remove as I go (in case as I read there might be pieces I want to keep.)

**Comment by shiffman** [comment]
PS: don't merge yet as I am going to comment inline on a few things!

File: content/10_nn.html [hunk 1]
```diff
@@ -10,33 +10,31 @@ <h1>Chapter 10. Neural Networks</h1>
   <a data-primary="artificial intelligence" data-type="indexterm"></a> <a
     data-primary="neural networks" data-type="indexterm"></a>
 
-  <p>We’re at the end of our story. This is the last official chapter of this book (though I
```
Comment (shiffman) [inline content/10_nn.html:13]: I might move this language / full paragraph to chapter 11 so maybe leave in for now or open an issue as a reminder?

File: content/10_nn.html [hunk 2]
```diff
@@ -60,10 +58,10 @@ <h2>10.1 Artificial Neural Networks: Introduction and Application</h2>
 
     <p>It’s probably pretty obvious to you that there are problems that are incredibly simple for a
       computer to solve, but difficult for you. Take the square root of 964,324, for example. A
-      quick line of code produces the value 982, a number Processing computed in less than a
+      quick line of code produces the value 982, a number your computer computed in less than a
       millisecond. There are, on the other hand, problems that are incredibly simple for you or me
       to solve, but not so easy for a computer. Show any toddler a picture of a kitten or puppy and
-      they’ll be able to tell you very quickly which one is which. Say hello and shake my hand one
+      they’ll be able to tell you very quickly which one is which. Say "hello" and shake my hand one
```
Comment (shiffman) [inline content/10_nn.html]:
```suggestion
      they’ll be able to tell you very quickly which one is which. Say “hello” and shake my hand one
```

**Comment by joeyklee** [comment]
Cool! I'll wait for your go!

File: content/10_nn.html [hunk 3]
```diff
@@ -427,65 +426,67 @@ <h2>10.3 Simple Pattern Recognition Using a Perceptron</h2>
...
     <p>The constructor could receive an argument indicating the number of inputs (in this case
       three: x, y, and a bias) and size the array accordingly.</p>
 
-    <pre data-code-language="java" data-type="programlisting" class="codesplit">
-  Perceptron(int n) {
-    weights = new float[n];
-    for (int i = 0; i < weights.length; i++) {
+    <pre data-code-language="javascript" data-type="programlisting" class="codesplit">
+  constructor(n) {
+    this.weights = new Array(n);
```
Comment (shiffman) [inline content/10_nn.html]:
```suggestion
    this.weights = [];
```

File: content/10_nn.html [hunk 4]
```diff
@@ -427,65 +426,67 @@ <h2>10.3 Simple Pattern Recognition Using a Perceptron</h2>
...
+    for (let i = 0; i < this.weights.length; i++) {
```
Comment (shiffman) [inline content/10_nn.html]:
```suggestion
    for (let i = 0; i < n; i++) {
```

File: content/10_nn.html [hunk 5]
```diff
@@ -805,46 +801,47 @@ <h2>10.4 Coding the Perceptron</h2>
...
   return 2*x+1;
 }
 
-void setup() {
-  size(640, 360);
+function setup() {
+  createCanvas(640, 360);
 
   ptron = new Perceptron(3);
+  training = new Array(2000);
```
Comment (shiffman) [inline content/10_nn.html]:
```suggestion
  training = [];
```

File: content/10_nn.html [hunk 6]
```diff
@@ -805,46 +801,47 @@ <h2>10.4 Coding the Perceptron</h2>
...
 
   //{!1} Make 2,000 training points.
-  for (int i = 0; i < training.length; i++) {
-    float x = random(-width/2,width/2);
-    float y = random(-height/2,height/2);
+  for (let i = 0; i < training.length; i++) {
```
Comment (shiffman) [inline content/10_nn.html]:
```suggestion
  for (let i = 0; i < 2000; i++) {
```

**Review by shiffman — APPROVED** [review]
Just some minor, nitpicky things, I started doing this but mostly I'll change and edit as I read over and rewrite the chapter! I am not 100% sure about my changes from `new Array(length)` to `[]` with `length` in the for loop, but I think that's more beginner friendly / typical?

File: content/10_nn.html (hunk 1 above)
Comment (joeyklee) [reply content/10_nn.html:13]: Sure!

Comment (joeyklee) [reply content/10_nn.html:13]: see https://github.com/nature-of-code/noc-book-2/issues/81

**Comment by joeyklee** [comment]
@shiffman - I went through and updated the rest of the code based on the suggestions above ^^. Thanks for reviewing! I will merge now.

**Pull request merged**